lib/src/git_backend.rs

```diff
@@ -491,6 +491,18 @@ impl Backend for GitBackend {
         let table = self.cached_extra_metadata_table()?;
         if let Some(extras) = table.get_value(id.as_bytes()) {
             deserialize_extras(&mut commit, extras);
+        } else {
+            let (table, table_lock) = self.read_extra_metadata_table_locked()?;
+            if let Some(extras) = table.get_value(id.as_bytes()) {
+                // Concurrent write_commit() would update extras before taking a lock.
+                deserialize_extras(&mut commit, extras);
+                *self.cached_extra_metadata.lock().unwrap() = Some(table);
+            } else {
+                // Make our change id persist (otherwise future write_commit() could reassign
+                // new change id.)
+                let extras = serialize_extras(&commit);
+                self.write_extra_metadata_entry(&table, &table_lock, id, extras)?;
```

Comment:
Yes, and the repo initialization will get slow. Perhaps, it can be mitigated by doing the import in bulk.
```sh
# original
% cd git
% time jj init --git-repo .
ignoring git submodule at "sha1collisiondetection"
Initialized repo in "."
jj init --git-repo .  1.93s user 0.27s system 99% cpu 2.209 total

# new
% time jj init --git-repo .
ignoring git submodule at "sha1collisiondetection"
Initialized repo in "."
jj init --git-repo .  3.25s user 5.12s system 97% cpu 8.547 total
```

Reply:
Yes, that would probably be a good idea, not only because it's faster, but also so we avoid creating tens of thousands of files (right?). Do you think we should get this merged now or do you want to fix that first?

Reply:
I'll make a follow-up PR. I think it can be done at `git::import_refs()` in a similar way to `repo.add_head()`.

Heavy jj user would already have ~10000 table files, so I don't think this PR is significantly bad.

Reply:
Sounds good. Thanks you!